Lib/test/test_dataclasses.py

```diff
@@ -1442,16 +1442,27 @@ class B:
                     replace(obj, x=0)
 
     def test_is_dataclass_genericalias(self):
```

Comment:
It is was test for very special corner case. It is very unlikely that such case (a dataclass which is a subclass of `types.GenericAlia`) occurs in real code. There is no guarantees that we will support such combination in future, so this test can be changed or removed. But your cases (generic dataclasses) are normal and common. I think that it would be better to add a separate test method for them.

Reply:
Fixed, moved to separate test